borg/archive.py

```diff
@@ -302,7 +305,7 @@ def save(self, name=None, timestamp=None):
             raise self.AlreadyExists(name)
         self.items_buffer.flush(flush=True)
         if timestamp is None:
-            self.end = datetime.utcnow()
+            self.end = self.start + timedelta(seconds=time.monotonic() - self.start_monotonic)
```

Comment:
Good point. Using the likely corrected time has the advantage that it's probably correct, otoh the start time wouldn't match the timestamps of any log entries associated with the job -- but then again they should be linked by the archive name.